sympy/solvers/tests/test_ode.py

```diff
@@ -1103,11 +1110,11 @@ def test_solve_ics():
     assert solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2],
         {f(0): 1, f(x).diff(x).subs(x, 0): 1}) == {C1: 1, C2: 1}
 
-    # XXX: Ought to be ValueError
-    raises(NotImplementedError, lambda: solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2], {f(0): 1, f(pi): 1}))
+    assert solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2],
+        {f(0): 1}) == {C2: 1}
 
     # XXX: Ought to be ValueError
-    raises(ValueError, lambda: solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2], {f(0): 1}))
+    raises(NotImplementedError, lambda: solve_ics([Eq(f(x), C1*sin(x) + C2*cos(x))], [f(x)], [C1, C2], {f(0): 1, f(pi): 1}))
```

Comment:
I think maybe this should return `{C2: 0}`.

Reply:
I didn't write that test case. It was already there. Well that looks like a new bug in solve_ics. I'll try to solve it.

Reply:
It looks like `solve` can handle this:
```julia
In [18]: C1, C2 = symbols('C1:3')                                                                                                                             

In [19]: sol = C1*sin(x) + C2*cos(x)                                                                                                                          

In [20]: sol                                                                                                                                                  
Out[20]: C₁⋅sin(x) + C₂⋅cos(x)

In [21]: eqs = [sol.subs(x, 0), sol.subs(x, pi)]                                                                                                              

In [22]: eqs                                                                                                                                                  
Out[22]: [C₂, -C₂]

In [23]: solve(eqs, [C1, C2])                                                                                                                                 
Out[23]: {C₂: 0}
```

Reply:
@oscarbenjamin why is C2 = 0 ? Shouldn't there'd be two values of C2, 1 and -1

Reply:
The equations are `C2=0` and `-C2=0` so we know that C2=0 but we can't say anything about C1.

Reply:
`f(0):1` means at `x=0` `f(0)=1` So putting values in this equation `f(x) = C1*sin(x) + C2*cos(x)` returns `C2=1`, If I am not making any mistake

Reply:
Oh sorry. You are right except that actually there are no solutions for C2:
```julia
In [1]: C1, C2 = symbols('C1:3')                                                                                                                               

In [2]: sol = C1*sin(x) + C2*cos(x)                                                                                                                            

In [3]: sol                                                                                                                                                    
Out[3]: C₁⋅sin(x) + C₂⋅cos(x)

In [4]: eqs = [sol.subs(x, 0)-1, sol.subs(x, pi)-1]                                                                                                            

In [5]: eqs                                                                                                                                                    
Out[5]: [C₂ - 1, -C₂ - 1]

In [6]: solve(eqs, [C1, C2])                                                                                                                                   
Out[6]: []
```
We don't have two values for C2 we have *no* values for C2 that are consistent with both initial conditions. Therefore we should raise ValueError in this case.

Reply:
So what can I do about the Travis-CI error ?

Reply:
This case should be ValueError